targets.py

```diff
@@ -2,34 +2,44 @@
 
 
 class DirectoryTarget:
-    def __init__(self, path):
-        self.name = path
+    def __init__(self, name, path):
+        self.name = name
+        self.path = path
 
     @property
     def id(self):
         return "Directory: " + self.name
 
+    @property
+    def mount_id(self):
+        return self.path
+
     def before_restore(self):
         pass
 
 
 class NamedVolumeTarget:
-    def __init__(self, name):
+    def __init__(self, name, volume):
         self.name = name
+        self.volume = volume
 
     @property
     def id(self):
         return "Named volume: " + self.name
 
+    @property
+    def mount_id(self):
+        return self.volume
+
     def _volume_exists(self):
         text = run(
             ["docker", "volume", "ls", "-q"],
             stdout=PIPE, universal_newlines=True
         ).stdout
         names = text.split('\n')
-        return self.name in names
+        return self.volume in names
 
     def before_restore(self):
         if not self._volume_exists():
-            print("Creating docker volume with name '{}'".format(self.name))
-            run(["docker", "volume", "create", "--name", self.name], stdout=PIPE)
+            print("Creating docker volume with name '{}'".format(self.volume))
+            run(["docker", "volume", "create", "--name", self.volume], stdout=PIPE)
```

Comment:
similarly, use

```python
self.docker.volumes.create(self.volume)
```

Reply:
Opened https://vimc.myjetbrains.com/youtrack/issue/VIMC-1590 for this